Client/Frontend/Share/ShareExtensionHelper.swift

```diff
@@ -60,36 +51,28 @@ class ShareExtensionHelper: NSObject {
             UIActivity.ActivityType.addToReadingList,
         ]
 
-        activityViewController.completionWithItemsHandler = { [weak self] activityType, completed, returnedItems, activityError in
-            guard let self = self else { return }
-            
-            print("Share type \(self.shareActivityType(activityType.map { $0.rawValue }))")
-            
-            if self.shareActivityType(activityType.map { $0.rawValue }) != .iBooks {
-                if !completed {
-                    completionHandler(completed, activityType, nil)
-                    return
-                }
-                // Bug 1392418 - When copying a url using the share extension there are 2 urls in the pasteboard.
-                // This is a iOS 11.0 bug. Fixed in 11.2
-                if UIPasteboard.general.hasURLs, let url = UIPasteboard.general.urls?.first {
-                    UIPasteboard.general.urls = [url]
-                }
-
-                completionHandler(completed, activityType, nil)
-            } else {
-                self.writeWebPagePDFDataToURL { url, error in
+        activityViewController.completionWithItemsHandler = { [weak selectedTab] activityType, completed, returnedItems, activityError in
+            if let activityType = activityType, activityType == .openInIBooks, let selectedTab = selectedTab {
+                // TODO: @JS Investigate why this is needed.
+                // Is it for Fix #2961? Fix sharing to iBooks
+                // (note: Could not test this as I could not trigger this scenario)
+                // #2961 steps don't use `openInIBooks`. Perhaps that's an os change?
+                Self.writeWebPagePDFDataToURL(selectedTab: selectedTab) { url, error in
                     completionHandler(completed, activityType, url)
                 }
+            } else {
+                completionHandler(completed, activityType, nil)
             }
         }
         
         return activityViewController
     }
-    
-    func writeWebPagePDFDataToURL(_ completion: @escaping (URL?, Error?) -> Void) {
+
+    // TODO: @JS Figure of if this is ever used
+    /// Function that writes the current webpage data to a PDF
+    private static func writeWebPagePDFDataToURL(selectedTab: Tab, completion: @escaping (URL?, Error?) -> Void) {
```

Comment:
Let's wait for @soner-yuksel review, then we can remove this method, i did not see it being called anywhere

Reply:
Yes we can absolutely remove this method.